nixos/modules/services/misc/bees.nix

```diff
@@ -0,0 +1,94 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  inherit (lib) mkOption types optionalString;
+  cfg = config.services.beesd;
+
+  fsOptions = {...}:
+  with types; {
+    options.spec = mkOption {
+      type = string;
+      description = ''
+        Specification usable by findmnt; ie. LABEL=foo, UUID=bar, etc.
+      '';
+    };
+    options.hashTableSizeMB = mkOption {
+      type = enum [ 16 32 64 128 256 512 1024 2048 4096 8192 16384 32768 65536 131072 262144 524288 1048576 2097152 4194304 8388608 ];
+      default = 1024; # 1GB; default from upstream beesd script
+      description = ''
+        Hash table size; must be 16MB*(2^n).
+
+        A larger ratio of index size to storage size means smaller blocks of duplicate content are recognized.
+
+        If you have 1TB of data, a 4GB hash table (which is to say, a value of 4096 -- 16MB*(2^8)) will permit 4KB extents (the smallest possible size) to be recognized, whereas a value of 1024 -- creating a 1GB hash table -- will recognize only aligned duplicate blocks of 16KB.
+      '';
+    };
+    options.verbosity = mkOption {
+      type = int;
+      default = 6;
+      description = "Log verbosity (0-8)";
+    };
+    options.workDir = mkOption {
+      type = str;
+      default = ".beeshome";
+      description = ''
+        Name (relative to the root of the filesystem) of the subvolume where the hash table will be stored.
+      '';
+    };
+  };
+
+in {
+
+  options.services.beesd = {
+    filesystems = mkOption {
+      description = "BTRFS filesystems to run block-level deduplication on";
+      default = { };
+      example = literalExample {
+        root = {
+          spec = "LABEL=root";
```

Comment:
Pretty sure this needs to be wrapped in a string literal for it to display correctly in the docs (`literalExample ''<nix code with newlines here>''`)